### PR TITLE
Fix active sync manager error on secondary master

### PR DESCRIPTION
**Pull request by bf8086**
There has been errors when an active sync enabled secondary master get promoted to primary. It is caused by active sync manager trying to access UFS when master is in secondary mode. UFS client are not initialized for mount points on secondary master so error occurs when active sync manager accesses it during snapshotting and server stopping process.

This fix updated active sync manager to avoid unnecessary access to UFS client on a secondary master.

### CODE REVIEW

**Comment by bf8086** [comment]
@yuzhu @gpang PTAL.

File: core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java [hunk 1]
```diff
@@ -164,6 +166,7 @@ public boolean isUnderSyncPoint(AlluxioURI path) {
    * Start the polling threads.
    */
   public void start() throws IOException {
+    mStarted = true;
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java:169]: Where is this set to `false`?

File: core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java [hunk 2]
```diff
@@ -428,8 +431,9 @@ public boolean hasNext() {
         while (mountId == -1) {
           try {
             syncPointPath = mEntry.getPath();
-            MountTable.Resolution resolution = mMountTable.resolve(mEntry);
-            mountId = resolution.getMountId();
+            String mountPoint = mMountTable.getMountPoint(mEntry);
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java:434]: What is the difference from before? How does the new code avoid UFS access?

**Review by gpang — COMMENTED** [review]
@bf8086 Thanks! I left a few questions

File: core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java (hunk 1 above)
Comment (bf8086) [reply core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java:169]: Added in `stop()`.

File: core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java (hunk 2 above)
Comment (bf8086) [reply core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java:434]: `MountTable::resolve()` will actually creates a UFS client to query the UFS path while the new code only grabs the `MountInfo`. This function only need the mount id so resolution is not required.

**Review by yuzhu — APPROVED** [review]
thanks for fixing this! LGTM after test passes

**Review by gpang — APPROVED** [review]
LGTM

**Comment by bf8086** [comment]
Alluxio-bot, merge this please.

**Comment by bf8086** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please.

**Pull request merged**